scipy/optimize/__init__.py

```diff
@@ -16,6 +16,7 @@
 
    minimize - Unified interface for minimizers of multivariate functions
    minimize_scalar - Unified interface for minimizers of univariate functions
+   minimize_constrained - Interface for minimizers of constrained functions
```

Comment:
I tried to improve the description. 

About the name: I will not worry too much about the name until we have choosen which route to take (integrate it with ``minimize`` or create an separete interface for constrained solvers). But I think I like ``minimize_constrained`` better than ``nlp``.